diff -ru zstd-1.5.7.orig/build/cmake/contrib/pzstd/CMakeLists.txt zstd-1.5.7/build/cmake/contrib/pzstd/CMakeLists.txt
--- zstd-1.5.7.orig/build/cmake/contrib/pzstd/CMakeLists.txt	2026-07-02 17:22:10.357142600 +0800
+++ zstd-1.5.7/build/cmake/contrib/pzstd/CMakeLists.txt	2026-07-02 17:23:23.579372700 +0800
@@ -30,10 +30,6 @@
 
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
-if (CMAKE_USE_PTHREADS_INIT)
   target_link_libraries(pzstd ${ZSTD_LIB} ${CMAKE_THREAD_LIBS_INIT})
-else()
-    message(SEND_ERROR "ZSTD currently does not support thread libraries other than pthreads")
-endif()
 
 install(TARGETS pzstd RUNTIME DESTINATION "bin")
